server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -115,7 +116,8 @@ private void setThreshold(float threshold) {
     @Override
     public void allocate(RoutingAllocation allocation) {
         if (allocation.routingNodes().size() == 0) {
-            /* with no nodes this is pointless */
+            // If no data node then set AllocationStatus to DECIDERS_NO
```

Comment:
No need for this comment if the method is better-named.

```suggestion
```

Reply:
Will resolve in next revision. Please review other changes